doc/source/development/dispatching.rst

```diff
@@ -0,0 +1,86 @@
+Dispatching
+===========
+
+The dispatching mechanism allows users to use alternative implementations of the algorithms
+available in scikit-image. The alternative implementations are provided by separate
+Python packages maintained outside of scikit-image. Example use-cases are support for array
+libraries other than Numpy or optimised implementations for particular hardware.
+
+This section of the documentation describes how to create an alternative implementation (a "backend")
+and how the dispatching mechanism in scikit-image works.
+
+
+Implementing a new backend
+--------------------------
+
+An alternative implementation ("backend") is a good place to provide optimized implementations
+for particular hardware, support array libraries other than Numpy or explore novel ideas that
+are not (yet) a good fit for the core scikit-image library.
+
+To create a backend you have to create a new Python package that registers two particular
+entrypoints. The first entrypoint called `skimage_backends` should resolve to a namespace
```

Comment:
Link to https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points here?